rfcs/2021-03-02-6600-vector-remap-language-web.md

```diff
@@ -0,0 +1,169 @@
+# RFC 6600 - 2021-03-03 - Vector Remap Language on the web
+
+From the beginning of our work on [Vector Remap Language][VRL] (VRL), we've had discussions about
+novel ways to make VRL approachable and even downright fun for users. Thus far, we've done some
+pretty unique things, like providing a REPL for the language. But we've also had plans, though
+fairly unspecific ones, about making VRL available on the web in some form. This RFC seeks to lend
+some concreteness to those discussions.
+
+## Options
+
+In this RFC, I discuss three possibities. The first two emerged from our internal discussions, while
+the third was gathered from a survey of the Wasm ecosystem.
+
+1. An in-browser [web app][web_app]/playground for experimenting with and learning VRL
+2. A [web server][web_server] that can process VRL programs and events
+3. A [published Wasm module], largely for internal use
+
+I propose pursuing #1 and #3 in the near term, while leaving #2 as an option for the future.
+
+### Interactive web app
+
+An interactive VRL web app would be a browser-based app that provides an aesthetically pleasing and
+UX-friendly experience for learning and experimenting with VRL. In its most basic form, it would
+enable you to enter a VRL event and a VRL program into a text area and see what happens. But there
+are other capabilities that we should consider:
+
```

Comment:
Just wondering if this be successful without the ability to select different versions of VRL?  Would this just be the latest released version?  Other related examples like the [rust playground](https://play.rust-lang.org) and [TS Playground](https://www.typescriptlang.org/play) allow you to select different versions including the nightly version. Of course that has implications for how we build this.

Reply:
My gut feeling is that the app should only have a concept of "latest" until we can establish that people are deriving real value from it

Reply:
> If you build it, they will come.

Yep, I hear ya.  I'm concerned we may not be giving this a fair go by being too conservative on features.   Also thinking about ways we can measure engagement.  That's something we could talk about next week.

Reply:
Regarding "being too conservative on features:" now *that* is the kind of can-do gumption we like around here :+1: